.github/actions/check_artifact_size/action.yaml

```diff
@@ -0,0 +1,140 @@
+            const workflowRuns = await github.rest.actions.listWorkflowRuns({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              workflow_id: '${{ inputs.workflow }}.yaml',
+              branch: 'main',
+              status: 'success',
+              per_page: 1
+            });
+
+            const latestRun = workflowRuns.data.workflow_runs[0].id;
+
+            // Get the artifact uploaded on the latest successful workflow run on the main branch.
+            const allArtifacts = await github.rest.actions.listWorkflowRunArtifacts({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              run_id: latestRun
+            });
+
+            const matchArtifacts = allArtifacts.data.artifacts.filter((artifact) => {
+              return artifact.name == '${{ inputs.name }}';
+            });
+
+            if (matchArtifacts.length == 1) {
+              console.log(`Found the latest uploaded artifact ${{ inputs.name }} on the main branch.`);
+
+              const download = await github.rest.actions.downloadArtifact({
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                artifact_id: matchArtifacts[0].id,
+                archive_format: 'zip',
+              });
+
+              const downloadDir = path.join(process.env.GITHUB_WORKSPACE, 'artifact_tmp');
+              fs.mkdirSync(downloadDir);
+              fs.writeFileSync(path.join(downloadDir, `${{ inputs.name }}.zip`), Buffer.from(download.data));
+
+              core.setOutput("downloadDir", downloadDir);
+            } else {
+              core.setFailed(`Expected one artifact with name ${{ inputs.name }}. Found ${matchArtifacts.length}.`);
+            }
+
+      - name: 'Unzip artifact from main branch'
+        id: unzip-downloaded-artifact
+        shell: bash
+        run: |
+          unzip "${{ steps.download-artifact.outputs.downloadDir }}/${{ inputs.name }}.zip" -d "${{ steps.download-artifact.outputs.downloadDir }}"
+
+      - name: 'Check new artifact size against main branch'
+        id: check-artifact-size
+        uses: actions/github-script@v6
+        with:
+          script: |
+            const fs = require('fs');
+            const path = require('path');
+
+            // Store file size thresholds in fileSizeThresholds.
+            const fileSizeThresholds = JSON.parse('${{ inputs.thresholds }}');
+
+            console.log(JSON.stringify(fileSizeThresholds, null, 2));
+
+            for (let file in fileSizeThresholds) {
+              console.log(`Checking file size of ${file}.`);
+
+              const downloadFilePath = path.join('${{ steps.download-artifact.outputs.downloadDir }}', file);
+              if (!fs.existsSync(downloadFilePath)) {
+                console.log(`File ${file} was not uploaded to the main branch.`);
+                continue;
```

Comment:
Do you expect one of the files to not ever exist?  Seems like it should fail here due to misconfiguration rather than just log.

Reply:
Yes, totally agree that error log is more suitable for cases of misconfiguration.